cli: TestZip failed

cli.TestZip [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3294900&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3294900&tab=artifacts#/) on master @ [de520ba29884af8fb9fa3700fd0a5b91f79afe55](https://github.com/cockroachdb/cockroach/commits/de520ba29884af8fb9fa3700fd0a5b91f79afe55):


```
I210812 20:50:51.582378 271714 sql/sqlliveness/slinstance/slinstance.go:168  [n1] 109  created new SQL liveness session e55f08cbcf7944a78a1085a6c75999f8
I210812 20:50:51.583357 271716 sql/temporary_schema.go:528  [n1] 110  found 0 temporary schemas
I210812 20:50:51.584322 271716 sql/temporary_schema.go:531  [n1] 111  early exiting temporary schema cleaner as no temporary schemas were found
I210812 20:50:51.585249 271716 sql/temporary_schema.go:532  [n1] 112  completed temporary object cleanup job
I210812 20:50:51.585826 271716 sql/temporary_schema.go:611  [n1] 113  temporary object cleaner next scheduled to run at 2021-08-12 21:20:51.534516951 +0000 UTC
W210812 20:50:51.731910 271403 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 114  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
I210812 20:50:51.732385 271424 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 115 ={"Timestamp":1628801451599151690,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
W210812 20:50:51.743483 271418 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 116  range lookup of key /Table/20 found only non-matching ranges []; retrying
I210812 20:50:51.820468 271988 migration/migrationmanager/manager.go:106  [n1,intExec=set-setting,migration-mgr] 117  no need to migrate, cluster already at newest version
I210812 20:50:51.892745 271988 util/log/event_log.go:32  [n1,intExec=set-setting] 118 ={"Timestamp":1628801451795475704,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.1-126'"],"SettingName":"version","Value":"21.1-126"}
I210812 20:50:52.233230 271935 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 119 ={"Timestamp":1628801452139978855,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"8a285acf-1799-42eb-b99b-433d60368e2d"}
I210812 20:50:52.353429 272408 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 120 ={"Timestamp":1628801452279359722,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210812 20:50:52.438704 272449 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 121 ={"Timestamp":1628801452359526667,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210812 20:50:52.865569 271323 server/server_sql.go:1036  [n1] 122  done ensuring all necessary startup migrations have run
I210812 20:50:52.867514 272571 jobs/job_scheduler.go:349  [n1] 123  waiting 3m0s before scheduled jobs daemon start
I210812 20:50:52.920549 272631 server/auto_upgrade.go:55  [n1] 124  no need to upgrade, cluster already at the newest version
I210812 20:50:53.297309 271565 1@gossip/gossip.go:1505  [n1] 125  node has connected to cluster via gossip
I210812 20:50:53.303154 271565 kv/kvserver/stores.go:269  [n1] 126  wrote 0 node addresses to persistent storage
I210812 20:50:53.409716 271630 3@vendor/github.com/cockroachdb/pebble/event.go:617  [n1,pebble,s1] 127  [JOB 3] WAL created 000004
I210812 20:50:53.505313 271323 1@server/server.go:2321  [n1] 128  serving sql connections
I210812 20:50:53.507210 271323 cli/testutils.go:139  [-] 129  server started at 127.0.0.1:37979
I210812 20:50:53.508001 271323 cli/testutils.go:140  [-] 130  SQL listener at 127.0.0.1:40557
[debug --host=127.0.0.1:37979 --insecure=false --certs-dir=/go/src/github.com/cockroachdb/cockroach/artifacts/cli-test627613521 zip --concurrency=1 --cpu-profile-duration=1s /dev/null]
I210812 20:51:01.396283 271675 2@util/log/event_log.go:32  [n1] 131 ={"Timestamp":1628801461396220337,"EventType":"runtime_stats","MemRSSBytes":1371484160,"GoroutineCount":252,"MemStackSysBytes":3604480,"GoAllocBytes":78625624,"GoTotalBytes":149874744,"GoStatsStaleness":0.005369537,"HeapFragmentBytes":16180392,"HeapReservedBytes":31547392,"HeapReleasedBytes":138477568,"CGoAllocBytes":20151872,"CGoTotalBytes":38080512,"CGoCallRate":0.00002376103,"CPUUserPercent":0.000048081976,"CPUSysPercent":0.0000045493575,"GCPausePercent":0.000000005470328,"GCRunCount":270,"NetHostRecvBytes":333158,"NetHostSendBytes":333158}
I210812 20:51:04.133001 271581 3@vendor/github.com/cockroachdb/pebble/event.go:617  [n1,pebble,s1] 132  [JOB 4] WAL created 000005
    zip_test.go:143: 
        testdata/zip/testzip:1: 
        output didn't match expected:
        @@ -18,11 +18,14 @@
         [cluster] retrieving SQL data for crdb_internal.jobs... writing output: debug/crdb_internal.jobs.txt... done
         [cluster] retrieving SQL data for system.jobs... writing output: debug/system.jobs.txt... done
         [cluster] retrieving SQL data for system.descriptor... writing output: debug/system.descriptor.txt... done
         [cluster] retrieving SQL data for system.namespace... writing output: debug/system.namespace.txt... done
         [cluster] retrieving SQL data for system.scheduled_jobs... writing output: debug/system.scheduled_jobs.txt... done
        -[cluster] retrieving SQL data for "".crdb_internal.create_statements... writing output: debug/crdb_internal.create_statements.txt... done
        +[cluster] retrieving SQL data for "".crdb_internal.create_statements... writing output: debug/crdb_internal.create_statements.txt...(error encountered after some results were delivered)
        +
        +[cluster] retrieving SQL data for "".crdb_internal.create_statements: last request failed: pq: query execution canceled due to statement timeout
        +[cluster] retrieving SQL data for "".crdb_internal.create_statements: creating error output: debug/crdb_internal.create_statements.txt.err.txt... done
         [cluster] retrieving SQL data for "".crdb_internal.create_type_statements... writing output: debug/crdb_internal.create_type_statements.txt... done
         [cluster] retrieving SQL data for crdb_internal.kv_node_liveness... writing output: debug/crdb_internal.kv_node_liveness.txt... done
         [cluster] retrieving SQL data for crdb_internal.kv_node_status... writing output: debug/crdb_internal.kv_node_status.txt... done
         [cluster] retrieving SQL data for crdb_internal.kv_store_status... writing output: debug/crdb_internal.kv_store_status.txt... done
         [cluster] retrieving SQL data for crdb_internal.regions... writing output: debug/crdb_internal.regions.txt... done
I210812 20:51:11.052037 271323 cli/testutils.go:205  [-] 133  stopping server and cleaning up CLI test
I210812 20:51:11.052788 271323 cli/testutils.go:169  [-] 134  stopping server at 127.0.0.1:37979 / 127.0.0.1:40557
W210812 20:51:11.060187 271714 sql/sqlliveness/slinstance/slinstance.go:207  [n1] 135  exiting heartbeat loop
W210812 20:51:11.061163 271728 jobs/registry.go:812  [-] 136  canceling all adopted jobs due to stopper quiescing
I210812 20:51:11.227842 271678 kv/kvclient/kvcoord/transport_race.go:106  [-] 137  transport race promotion: ran 78 iterations on up to 1906 requests
--- FAIL: TestZip (20.52s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestZip PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/server knz
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -121,6 +121,8 @@ ORDER BY name ASC`)
 func TestZip(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
+	skip.UnderRace(t, "test too slow under race")
+
 	dir, cleanupFn := testutils.TempDir(t)
 	defer cleanupFn()
 

